ml-proto/test/left-to-right.wast

```diff
@@ -0,0 +1,173 @@
+(module
+  (memory 1)
+  
+  ;; The idea is: We reset the memory, then the arithmetic instruction calls $*_left and $*_right.
+  ;; $*_left stores the number 1, and $*_right stores the number 2 (both at address 0)
+  ;; Then we read the value at address 0. If it's 0, the VM did an unwanted optimzation.
+  ;; If it's 1, then $*_right was called before $*_left. If it's 2, then everything's fine.
+  (func $reset (i32.store8 (i32.const 0)(i32.const 0)))
+  (func $get (result i32) (i32.load8_u (i32.const 0)))
+  (func $i32_left (result i32) (i32.store8 (i32.const 0) (i32.const 1)) (i32.const 127))
+  (func $i32_right (result i32) (i32.store8 (i32.const 0) (i32.const 2)) (i32.const 141))
+  (func $i64_left (result i64) (i32.store8 (i32.const 0) (i32.const 1)) (i64.const 163))
+  (func $i64_right (result i64) (i32.store8 (i32.const 0) (i32.const 2)) (i64.const 155))
+  (func $f32_left (result f32) (i32.store8 (i32.const 0) (i32.const 1)) (f32.const 124))
+  (func $f32_right (result f32) (i32.store8 (i32.const 0) (i32.const 2)) (f32.const 145))
+  (func $f64_left (result f64) (i32.store8 (i32.const 0) (i32.const 1)) (f64.const 163))
+  (func $f64_right (result f64) (i32.store8 (i32.const 0) (i32.const 2)) (f64.const 164))
+
+
+  (func $i32_add (result i32) (call $reset)(i32.add (call $i32_left)(call $i32_right))(call $get))
```

Comment:
For consistency with the overall style of the tests in this directory, please put spaces between operands.